homeassistant/components/google_cloud/stt.py

```diff
@@ -0,0 +1,147 @@
+"""Support for the Google Cloud STT service."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator, AsyncIterable
+import logging
+
+from google.api_core.exceptions import GoogleAPIError, Unauthenticated
+from google.cloud import speech_v1
+
+from homeassistant.components.stt import (
+    AudioBitRates,
+    AudioChannels,
+    AudioCodecs,
+    AudioFormats,
+    AudioSampleRates,
+    SpeechMetadata,
+    SpeechResult,
+    SpeechResultState,
+    SpeechToTextEntity,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    CONF_SERVICE_ACCOUNT_INFO,
+    CONF_STT_MODEL,
+    DEFAULT_STT_MODEL,
+    DOMAIN,
+    STT_LANGUAGES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Google Cloud speech platform via config entry."""
+    service_account_info = config_entry.data[CONF_SERVICE_ACCOUNT_INFO]
+    client = speech_v1.SpeechAsyncClient.from_service_account_info(service_account_info)
+    async_add_entities([GoogleCloudSpeechToTextEntity(config_entry, client)])
+
+
+class GoogleCloudSpeechToTextEntity(SpeechToTextEntity):
+    """Google Cloud STT entity."""
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        client: speech_v1.SpeechAsyncClient,
+    ) -> None:
+        """Init Google Cloud STT entity."""
+        self._attr_unique_id = f"{entry.entry_id}-stt"
```

Comment:
I copied this from other STT entities such as https://github.com/home-assistant/core/blob/bb5640b41b414064907046033321e9ae1b5a6bbd/homeassistant/components/wyoming/stt.py#L56
If you are certain it would be fine and you prefer it without the suffix, let me know to remove it before the next release.

Reply:
Yes I'm certain. What we do need is to deduplicate each entity within the same platform type. Eg if two different config entries are used that creates two different STT entities, we need to have an id for that to deduplicate them (config entry id). If one platform creates two entities for every config entry we need to have another id for that.

Reply:
Addressing it in #126585